wbb/modules/sudoers.py

```diff
@@ -135,20 +137,41 @@ async def get_stats(_, message: Message):
     filters.user(SUDOERS) & cust_filter.command(commands=("song"))
 )
 async def song(_, message: Message):
-    text = message.text.replace("/song ", "")
-    query = text.replace(" ", "%20")
-    r = requests.get(f"{JSMAPI}{query}")
-
-    i = 0
-    while i < 2:
-        sname = r.json()[i]['song']
-        slink = r.json()[i]['media_url']
-        ssingers = r.json()[i]['singers']
-        sduration = r.json()[i]['duration']
-        file = wget.download(slink)
-        ffile = file.replace("mp4", "m4a")
-        os.rename(file, ffile)
-        await message.reply_audio(audio=ffile, title=sname,
-                                  performer=ssingers, duration=int(sduration))
-        i += 1
-        os.remove(ffile)
+    text = message.text.replace("/song", "")
+    query = text.replace(" ", "")
+    if text != "":
+        m = await message.reply_text("Searching...")
+        r = requests.get(f"{JSMAPI}{query}")
+
+        i = 0
+        while i < 2:
+            sname = r.json()[i]['song']
+            slink = r.json()[i]['media_url']
+            ssingers = r.json()[i]['singers']
+            file = wget.download(slink)
+            ffile = file.replace("mp4", "m4a")
+            os.rename(file, ffile)
+            await message.reply_audio(audio=ffile, title=sname,
+                                      performer=ssingers)
+            i += 1
+            os.remove(ffile)
+        await m.delete()
+    else:
+        await message.reply_text("/song requires an argument.")
+# Upload
+
+
+@app.on_message(
+    filters.user(SUDOERS) & cust_filter.command(commands=("upload"))
+)
+async def upload(_, message: Message):
+    urli = message.text.replace("/upload", "")
+    url = urli.replace(" ", "")
+    if url != "":
+        data = BytesIO(requests.get(url).content)
+        data.name = url.split("/")[-1]
+        m = await message.reply_text("Uploading.....")
+        await message.reply_document(data)
+        await m.delete()
+    else:
+        await message.reply_text("/upload requires an argument")
```

Comment:
maybe you can check length of the list of command

```
if len(message.command) = 1:
    await message.reply_text("/upload requires an argument")
    return
url = message.command.[1]
...
```

the disadvantage is that the if the url contains a space then it won't work, but the url shouldn't contain a space, it should be encoded before user send the command.